converter/dicom-cast/Table/Registration/DiocmCastTableRegistrationExtension.cs

```diff
@@ -0,0 +1,98 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Health.DicomCast.Core.Features.TableStorage;
+using Microsoft.Health.DicomCast.TableStorage.Configs;
+using Microsoft.Health.DicomCast.TableStorage.Features.Health;
+using Microsoft.Health.DicomCast.TableStorage.Features.Storage;
+using Microsoft.Health.Extensions.DependencyInjection;
+
+namespace Microsoft.Health.DicomCast.TableStorage
+{
+    public static class DiocmCastTableRegistrationExtension
+    {
+        public const string TableStoreConfigurationSectionName = "TableStore";
+
+        /// <summary>
+        /// Adds the table data store for dicom cast.
+        /// </summary>
+        /// <param name="serviceCollection">Service collection</param>
+        /// <param name="configuration">The configuration for the server.</param>
+        /// <returns>IServiceCollection</returns>
+        public static IServiceCollection AddTableStorageDataStore(this IServiceCollection serviceCollection, IConfiguration configuration)
+        {
+            EnsureArg.IsNotNull(serviceCollection, nameof(serviceCollection));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            if (configuration.GetSection("TableStore").GetSection("Enabled").Get<bool>() == true)
+            {
+                serviceCollection
+                        .AddTableDataStore(configuration)
+                        .AddHealthChecks().AddCheck<TableHealthCheck>(name: nameof(TableHealthCheck));
+
+                serviceCollection.Add<TableExceptionStore>()
+                    .Singleton()
+                    .AsSelf()
+                    .AsImplementedInterfaces();
+            }
+            else
+            {
+                serviceCollection.Add<NullExceptionStore>()
+                    .Singleton()
+                    .AsSelf()
+                    .AsImplementedInterfaces();
```

Comment:
Something to think about is that it may make sense to do this in a different extension like `AddDefaultExceptionStore` and if they have table storage enabled to replace the implementation of `IExceptionStore` with the table storage version.

Reply:
Updated, thanks